README.md

```diff
@@ -2,21 +2,177 @@
+* stores the events in external storage for presenting in the REST API.
+The SSE Server of Sidecar is optional. 
+
+#### REST API Server
+```mermaid
+   graph LR;
+   CLIENT{Client}
+   CLIENT --> REST_API
+   STORAGE[("Storage")]
+   REST_API --> STORAGE
+   CONFIG{{"Config file (toml)"}}
+   MAIN --1.reads--> CONFIG
+   subgraph "Casper sidecar"
+      MAIN[main.rs]
+      MAIN --2.spawns--> REST_API
+      REST_API["REST API"]
+   end
+```
+
+Sidecar offers a REST API that allows clients to query the events stored in external storage. The REST API is optional. The specific endpoints of the REST API is discoverable to node operators. See [OpenAPI](#openapi-specification) and [Swagger](#swagger-documentation) sections. Also, see the [usage instructions](USAGE.md) for details.
+
+#### ADMIN API Server
+```mermaid
+   graph LR;
+   CLIENT{Client}
+   CLIENT --> ADMIN_API
+   CONFIG{{Config file}}
+   MAIN --1.reads--> CONFIG
+   subgraph "Casper sidecar"
+      MAIN[main.rs]
+      MAIN --2.spawns--> ADMIN_API
+      ADMIN_API["ADMIN API"]
+   end
+```
+
+Sidecar offers an ADMIN API that allows the operator to look into the current operation of Sidecar. The ADMIN API is optional. For details please see the [admin server configuration](#admin-server).
+
+#### RPC API Server
+```mermaid
+   graph LR;
+   CLIENT{Client}
+   CLIENT --> RPC_API
+   CONFIG{{Config file}}
+   MAIN --1.reads--> CONFIG
+   CASPER_NODE(("Casper Node binary port"))
+   RPC_API --forwards request--> CASPER_NODE
+   subgraph "Casper sidecar"
+      MAIN[main.rs]
+      MAIN --2.spawns--> RPC_API
+      RPC_API["RPC JSON API"]
+   end
+```
+Sidecar offers an RPC JSON API. This module is optional. It is a json bridge between the end users and the binary port od casper node. The RPC API server forwards the requests to the Casper Node binary port. 
+For more details on how the RPC JSON API works please see the [rpc sidecar README](rpc_sidecar/README.md).
+
+To discuss the configuration of RPC API let's consider the following configuration section:
+
+```
+[rpc_server.main_server]
+enable_server = true
+address = '0.0.0.0:7777'
+qps_limit = 100
+max_body_bytes = 2_621_440
+cors_origin = ''
+
+[rpc_server.node_client]
+address = '127.0.0.1:28101'
+max_request_size_bytes = 4_194_304
+max_response_size_bytes = 4_194_304
+request_limit = 3
+request_buffer_size = 16
+
+[rpc_server.speculative_exec_server]
+enable_server = true
+address = '0.0.0.0:7778'
+qps_limit = 1
+max_body_bytes = 2_621_440
+cors_origin = ''
+
+
+[rpc_server.node_client.exponential_backoff]
+initial_delay_ms = 1000
+max_delay_ms = 32_000
+coefficient = 2
+max_attempts = 30
+```
+
+* `main_server.enable_server` - If set to true, the RPC API server will be enabled
+* `main_server.address` - Address under which the main RPC API server will be available
+* `main_server.qps_limit` - The maximum number of requests per second
+* `main_server.max_body_bytes` - Maximum body size of request to API in bytes
```

Comment:
Add a "." after each bullet point to be consistent with the rest of the document.
```suggestion
* `main_server.enable_server` - The RPC API server will be enabled if set to true.
* `main_server.address` - Address under which the main RPC API server will be available.
* `main_server.qps_limit` - The maximum number of requests per second.
* `main_server.max_body_bytes` - Maximum body size of request to API in bytes.
```